File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
      if(n==1){
        return n;
      }
    int f=1,s=2;
    for(int i=2;i<n;i++){
        int t=f+s;
        f=s;
        s=t;
    }
    return s;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp=[-1 for _ in range(n)]
        def solve(i):
            if i==1 or i==0:
                return 1
            if dp[i-1]!=-1:
                return dp[i-1]
            
            dp[i-1]=solve(i-1)+solve(i-2)
            return dp[i-1]
        return solve(n)